static/js/tools/choropleth/choropleth.tsx

```diff
@@ -0,0 +1,351 @@
+            .selectAll("path")
+            .data(geojson.features);
+        
+        // Create new infill.
+        mapContent
+            .attr("fill", function 
+                    (d : { properties: { geoId : string, pop : number }}) {
+                if (d.properties.geoId in values) {
+                    const value = values[d.properties.geoId];
+                    if (isPerCapita) {
+                        if (d.properties.hasOwnProperty("pop")) {
+                            return colorScale(value / d.properties.pop);
+                        }
+                        return "gray";
+                    }
+                    return colorScale(value);
+                } else {
+                    return "gray";
+                }
+            });
+    }
+
+    /**
+     * Capture hover event on geo and displays relevant information.
+     * @param {json} geo is the geoJson content for the hovered geo.
+     */
+    handleMapHover(geo) {
+        // Display statistical variable information on hover.
+        const name = geo.properties.name;
+        const geoId = geo.properties.geoId;
+        const values = this.state["values"];
+        let geoValue : any = "No Value";
+        if (geoId in values) {
+            geoValue = values[geoId];
+            if (this.state["perCapita"]) {
+                if (geo.properties.hasOwnProperty("pop")) {
+                    geoValue /= geo.properties.pop;
+                }
+            }
+        }
+
+        document.getElementById("hover-text-display").innerHTML =
+            name + " - " + geoValue.toLocaleString();        
+    
+        // Highlight selected geo in black on hover.
+        d3.select(geo.ref).attr("class", "border-highlighted");
+    }
+
+    handleStatVarChange(statVar) {
+        // TODO(iancostello): Update browser.
+        var baseUrl = buildChoroplethUrl(
+            ["geoId", "bc", "perCapita", "level", "mdom"],
+            "",
+        );
+        baseUrl += "&statVar=" + statVar;
+        history.pushState({}, null, baseUrl);
+        this.loadValues();
+    }
+  
+    /**
+     * Clears output after leaving a geo.
+     */
+    mouseLeave(geo) {
+        // Remove hover text.
+        document.getElementById("hover-text-display").innerHTML = "";
+    
+        // Remove geo display effect.
+        d3.select(geo.ref).attr("class", "border");
+    }
+    
+    /**
+     * Capture click event on geo and zooms
+     * user into that geo in the choropleth tool.
+     * @param {json} geo is the geoJson content for the clicked geo.
+     */
+    handleMapClick(geo) {
+        if (geo.properties.hasSublevel) {
+            redirectToGeo(geo.properties.geoId);
+        } else {
+            //TODO(iancostello): Improve this feature (change cursor)
+            alert("This geo has no further sublevels!");
+        }
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <svg id="map_container" width="800px" height="500px">
```

Comment:
Add a todo to capture the element's size instead of hardcoding.

Reply:
Added